src/ravindra_J/Assignment_No_14_Scanner_Class/StockUsingIfElseLadder.java

```diff
@@ -0,0 +1,174 @@
+/*1. Create a system (Shop) which maintains each item's quantity.
+a. Initial stocks of Maggie: 50 Maggie packets 
+b. Initial stocks of Dosa : 43 packets 
+c. Initial stocks of pouches : 39 oil packets 
+d. Initial stocks of panipuri : 43 packets  
+e. Initial stocks of masala�s : 73 packets.
+Expectation:
+1) Only one Customer came for shopping, he picked different items with different quantities,
+print "running out of stock" message if a particular item is sold out otherwise deduct requested 
+quantity from total quantity. Partial selling is allowed.  
+2) Create a method which shows only those item which is "Out of stock".
+3) Create a method which shows only those item which is "Available in stock" & respective quantities.*/
+
+package ravindra_J.Assignment_No_14_Scanner_Class;
+
+import java.util.Scanner;
+
+public class StockUsingIfElseLadder {
+
+
+	int initialMaggieStock;
+	int initialDosaStock;
+	int initialOilPacketsStock;
+	int initialPanipuriStock;
+	int initialMasalaStock;
+
+	public void setInitialStock(int maggieeQty,int dosaQty,int oilPQty,int panipuriPQty,int masalaPQty) {
+
+		initialMaggieStock = maggieeQty;
+		initialDosaStock = dosaQty;
+		initialOilPacketsStock = oilPQty;
+		initialPanipuriStock = panipuriPQty; 
+		initialMasalaStock = masalaPQty;
+	} 
+
+	public void inStock() {
+
+		if(initialMaggieStock > 0) 
+			System.out.println("\nAvailable Maggiee Stock :" +initialMaggieStock);
+		if(initialDosaStock>0)
+			System.out.println("Available Dosa Stock :" +initialDosaStock);
+		if(initialOilPacketsStock>0)
+			System.out.println("Available Oil Packets Stock :" +initialOilPacketsStock);
+		if(initialPanipuriStock>0)
+			System.out.println("Available Panipuri Packets Stock :" +initialPanipuriStock);
+		if(initialMasalaStock>0)
+			System.out.println("Available Masala Packets Stock : " +initialMasalaStock);
+	}
+
+
+	public void outOfStock() {
+
+		if(initialMaggieStock == 0) 
+			System.out.println("\nMaggiee Stock Currently not Available : " +initialMaggieStock);
+		if(initialDosaStock==0)
+			System.out.println("Dosa Stock Currently not Available : " +initialDosaStock);
+		if(initialOilPacketsStock==0)
+			System.out.println("Oil Packets Stock Currently not Available : " +initialOilPacketsStock);
+		if(initialPanipuriStock==0)
+			System.out.println("Panipuri Packets Stock Currently not Available : " +initialPanipuriStock);
+		if(initialMasalaStock==0)
+			System.out.println("Masala Packets Stock Currently not Available : " +initialMasalaStock);
+	}
+
+	public void pickDifferentItems(int reqMQ, int reqDQ, int reqOQ, int reqPQ, int reqMpQ,boolean partialQtyAllow) {
+
+		if(reqMQ <= initialMaggieStock ) {
+			initialMaggieStock = initialMaggieStock - reqMQ;
+			System.out.println("\n Maggie You are Requested : "+reqMQ+" and Updated Maggie Available Stock Is: " +initialMaggieStock);
+		}
+		else if(reqMQ >= initialMaggieStock && partialQtyAllow) {
```

Comment:
you already allow customer to purchase partial in **if condition** right?
then in else we should say that we can process with this stock only.